src/store/inbox-agent/inboxAgentSlice.ts

```diff
@@ -0,0 +1,41 @@
+import { createSlice, createEntityAdapter } from '@reduxjs/toolkit';
+import type { Agent } from '@/types';
+import { inboxAgentActions } from './inboxAgentActions';
+
+export const inboxAgentAdapter = createEntityAdapter<Agent>({
+  selectId: agent => agent.id,
+});
+
+export interface InboxAgentState {
+  uiFlags: {
+    isLoading: boolean;
+  };
+}
+
+const initialState = inboxAgentAdapter.getInitialState<InboxAgentState>({
+  uiFlags: {
+    isLoading: false,
+  },
+});
+
+const inboxAgentSlice = createSlice({
+  name: 'inboxAgent',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder
+      .addCase(inboxAgentActions.fetchInboxAgents.pending, state => {
+        state.uiFlags.isLoading = true;
+      })
+      .addCase(inboxAgentActions.fetchInboxAgents.fulfilled, (state, action) => {
+        const { payload: agents } = action.payload;
+        inboxAgentAdapter.setAll(state, agents);
```

Comment:
If the modeling is intended to store assignable agents in a single inbox, let’s not do that. Instead, let’s store it against the inbox ID (so that we don't have to fetch the assignable agents every time again).